### ISSUE TITLE
Error when change subtitle track 

### ISSUE THREAD

**Issue by xhbadxx** [google/ExoPlayer#9795]
Case i'm using ExoPlayer v2.16.1:
1. Change subtitle from video. 
2. Change bitrate or new video url..
--> Error and player can't play the video.

```
E/EventLogger: playerFailed [eventTime=962.41, mediaPos=38.14, window=0, period=0, errorCode=ERROR_CODE_FAILED_RUNTIME_CHECK
      com.google.android.exoplayer2.ExoPlaybackException: Unexpected runtime error
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:612)
        at android.os.Handler.dispatchMessage(Handler.java:103)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: java.lang.IllegalStateException
        at com.google.android.exoplayer2.util.Assertions.checkState(Assertions.java:84)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:314)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:259)
        at com.google.android.exoplayer2.MediaPeriodHolder.handlePrepared(MediaPeriodHolder.java:193)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handlePeriodPrepared(ExoPlayerImplInternal.java:2209)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:505)
        at android.os.Handler.dispatchMessage(Handler.java:103) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.os.HandlerThread.run(HandlerThread.java:67) 
    ]
```

Please help me to check this problem.

**Comment by marcbaechinger** [comment]
Assigning to @icbaker for subtitle expertise.

@xhbadxx I think we need some more info from your side though. I think it would be helpful to have the stream URL that you are playing. Also it would be helpful to tell us a bit on how you expect that the subtitle track is going to be enabled and if you did this in code please send us a snippet that tells what you did. 

The standard answer for reference: 

Please provide complete information as requested in the issue template. The issue template can be found [here](https://github.com/google/ExoPlayer/blob/release-v2/.github/ISSUE_TEMPLATE/bug.md). If you're unable to share bug reports or test content publicly, please send them to dev.exoplayer@gmail.com using a subject in the format "Issue #9795". Please also update this issue to indicate you’ve done this.

**Comment by sixones** [comment]
@marcbaechinger Seeing this as well with subtitles enabled when switching to an IMA ad pod, looks like Exo v2.16.1 is attempting to apply the track selection to the new content (in my case an advert) when the same tracks don't exist.

**Comment by icbaker** [comment]
@sixones I'm unable to reproduce a crash using version 2.16.1 when entering an IMA ad pod - but I did reproduce a crash with the same stack trace as this issue. I did the following:

1. Modified the demo app JSON to create an item with both both `ad_tag_uri` and `subtitle_uri`, `subtitle_mime_tye` and `subtitle_language`:
    ```
    {
      "name": "VMAP pre-, mid- and post-rolls, single ads",
      "uri": "https://storage.googleapis.com/exoplayer-test-media-1/mkv/android-screens-lavf-56.36.100-aac-avc-main-1280x720.mkv",
      "ad_tag_uri": "https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/ad_rule_samples&ciu_szs=300x250&ad_rule=1&impl=s&gdfp_req=1&env=vp&output=vmap&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ar%3Dpremidpost&cmsid=496&vid=short_onecue&correlator=",
      "subtitle_uri": "https://storage.googleapis.com/exoplayer-test-media-1/ttml/netflix_japanese_ttml.xml",
      "subtitle_mime_type": "application/ttml+xml",
      "subtitle_language": "ja"
    },
    ```
2. Played the modified media item.
3. Watched the pre-roll, content and first mid-roll with no crash.
4. I then repeated 2 & 3 but ensured I explicitly selected the subtitle track in between the pre-roll and mid-roll. I used the top track selector in the demo app ('SELECT TRACKS' button).
5. I then repeated 2 & 3 but selected the None subtitle track using the CC track selection button at the bottom of the demo app (built into `StyledPlayerView`). Then I selected the 'Japanese' subtitle track again. Playback crashed:
    ```
    2021-12-21 15:29:45.049 27812-27812/com.google.android.exoplayer2.demo E/EventLogger: playerFailed [eventTime=18.59, mediaPos=7.72, window=0, period=0, errorCode=ERROR_CODE_FAILED_RUNTIME_CHECK
      com.google.android.exoplayer2.ExoPlaybackException: Unexpected runtime error
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:612)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:245)
        at android.os.HandlerThread.run(HandlerThread.java:67)
     Caused by: java.lang.IllegalStateException
        at com.google.android.exoplayer2.util.Assertions.checkState(Assertions.java:84)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:314)
        at com.google.android.exoplayer2.MediaPeriodHolder.applyTrackSelection(MediaPeriodHolder.java:259)
        at com.google.android.exoplayer2.MediaPeriodHolder.handlePrepared(MediaPeriodHolder.java:193)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handlePeriodPrepared(ExoPlayerImplInternal.java:2209)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:505)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:245) 
        at android.os.HandlerThread.run(HandlerThread.java:67) 
    ```

I then repeated step 5 on the dev-v2 branch and was unable to reproduce the crash - so I think this has been fixed since 2.16.1 was released. It may have been fixed by https://github.com/google/ExoPlayer/commit/8c90ba5db462ee0c1bfadd67a817092ee23af0eb or other recent track selection changes by @tonihei but I'm not sure.

@xhbadxx Are you able to reproduce the crash using the demo app built from the dev-v2 branch?

**Comment by dgelormini** [comment]
I'm running into this issue as well with the 2.16.1; easiest way to repro on my end is to just select a different CC option then pause/resume the app. Tested on the `dev-v2` branch and it appears to be working there.
